src/AttrValidator.hack

```diff
@@ -95,10 +95,10 @@ class HTMLPurifier_AttrValidator {
                     // that must be overridden.
                     // Theoretically speaking, we could have a
                     // AttrDef_DenyAll, but this is faster!
-                    $result = false;
+                    $result = null;
                 } else {
                     // validate according to the element's definition
-                    $value = (string)$value;
+                    $value = $value;
```

Comment:
If we're no longer typecasting we don't need this line anymore (same with line 111)